src/abstractdataframe/prettytables.jl

```diff
@@ -0,0 +1,139 @@
+
+    # Assemble the configurations for this print.
+    _local_conf = deepcopy(_pretty_tables_conf)
+
+    for kw in kwargs
+        _local_conf[kw[1]] = kw[2]
+    end
+
+    # Check if the user wants to display a summary about the DataFrame that is
+    # being printed. This will be shown using the `title` option of
+    # `pretty_table`.
+    title = summary ? Base.summary(df) : ""
+
+    # Transform into a named tuple so that it can be passed to PrettyTables.jl.
+    dictkeys = (collect(keys(_local_conf))...,)
+    dictvals = (collect(values(_local_conf))...,)
+    nt = NamedTuple{dictkeys}(dictvals)
+
+    # Print the table with the selected options.
+    try
+        pretty_table(io, df,
+                     vcat(names,types);
+                     crop = crop,
+                     nosubheader = !eltypes,
+                     row_number_column_title = string(rowlabel),
+                     title = title,
+                     nt...)
+    catch
+        @warn """An unsupported argument was passed to PrettyTables.jl.
+                 The default configuration will be used."""
+
+        dictkeys = (collect(keys(_pretty_tables_safeconf))...,)
+        dictvals = (collect(values(_pretty_tables_safeconf))...,)
+        nt_sc = NamedTuple{dictkeys}(dictvals)
+
+        pretty_table(io, df,
+                     vcat(names,types);
+                     crop = crop,
+                     maximum_columns_width = truncstring,
+                     nosubheader = !eltypes,
+                     row_number_column_title = string(rowlabel),
+                     title = title,
+                     nt_sc...)
+    end
+end
+
+################################################################################
+#                             Auxiliary functions
+################################################################################
+
+# Default DataFrames highlighter for text backend.
+#
+# This highlighter change the text color to gray in cells with `nothing`,
+# `missing`, `#undef`, and types related to DataFrames.jl.
+function _df_h_f(data,i,j)
+    try
+        return ismissing(data[i,j]) ||
+            data[i,j] == nothing ||
+            typeof(data[i,j]) <: Union{AbstractDataFrame, GroupedDataFrame,
+                                       DataFrames.DataFrameRow,
+                                       DataFrames.DataFrameRows,
+                                       DataFrames.DataFrameColumns}
+    catch e
+        if isa(e, UndefRefError)
+            return true
+        else
+            rethrow(e)
+        end
+    end
+end
+
+# Default DataFrames formatter for text backend.
+#
+# This formatter changes how the following types are presented when rendering
+# the data frame:
+#     - missing;
+#     - nothing;
+#     - Cells with types related to DataFrames.jl.
+
+function _df_f(v,i,j)
+    if typeof(v) <: Union{AbstractDataFrame, GroupedDataFrame,
+                          DataFrames.DataFrameRow, DataFrames.DataFrameRows,
```

Comment:
no need of `DataFrames.` prefix

Reply:
Yes! Good catch. This came from that code snippet.